confidence/models.py

```diff
@@ -42,15 +46,14 @@ def __init__(self, *sources, separator='.', missing=Missing.silent):
         :param missing: policy to be used when a configured key is missing,
             either as a `.Missing` instance or a default value
         """
-        self._separator = separator
-        self._missing = missing
-        self._root = self
+        self._separator: str = separator
+        self._missing: typing.Any = missing
+        self._root: Configuration = self
```

Comment:
Are these type hints necessary for mypy? I would hope that they can be directly inferred from the type hints you already provided in the function definition.

Reply:
I see that `self._missing: typing.Any` is required by mypy, but the other two aren't I think.

Reply:
I see that for `self._missing` mypy does require the explicit typehint, but not for the other two...

Reply:
Ah, yes, that might even be a lot more simple now we've added `allow_redefinition` :thinking:

Reply:
Made all the args typed as `Missing`, though that seems not entirely true, as the docs state that either a `Missing` instance or a default value is accepted... That would make it `Union[Missing, Any]`, which is kinda silly, really, so ... just `Any`?

Reply:
Yeah if there are no further restrictions on the default value, `Any` seems like the only logical choice.